### PR TITLE
Update MSRV to 1.56.1

### PR DESCRIPTION
**Pull request by storopoli**
Following `rust-bitcoin` MSRV of 1.56.1, this pins MSRV to 1.56.1.

Quoting a recent merge in `rust-bitcoin`

> Rust version 1.56.0 introduced edition 2021. Shortly afterwards, on
> October 21 2021 Rust version 1.56.1 was released.
>
> Debian stable is currently shipping `rustc 1.63.0`.
>
> Our stated MSRV policy is: In Debian stable and at least 2 years old.
>
> Therefore our MSRV policy is met by Rust version 1.56.1 and we can strat
> to bump our MSRV org wide.
>
> Links:
> - https://blog.rust-lang.org/2021/11/01/Rust-1.56.1.html
> - https://blog.rust-lang.org/2021/10/21/Rust-1.56.0.html
> - https://packages.debian.org/stable/rust/rustc

Removing the pins in `contrib/test.sh` since they are not necessary
in 1.56.1.

EDIT: CI fails because of the `#[allow(dead_code)]` implemented in #638 on some Psbt traits.
Do I port these to here? Conceptually they belong to #638 and not here.

### CODE REVIEW

File: .github/workflows/rust.yml
Comment (storopoli) [inline .github/workflows/rust.yml:57]: Always quote numbers in YAML.

See https://ruudvanasseldonk.com/2023/01/11/the-yaml-document-from-hell

File: contrib/test.sh
Comment (apoelstra) [inline contrib/test.sh:28]: In ad446c23275cf2c97e993102014c8498c69f01f3:

Are these pins actually necessary?

Can you try removing the pins, compiling with 1.56.1, and then re-adding them as necessary?

Comment (storopoli) [reply contrib/test.sh]: You were right. They are not necessary. I've removed them in c301ce3688e66a1f0f041154c4f2455c9dfa10e4.

**Review by apoelstra — APPROVED** [review]
ACK c301ce3688e66a1f0f041154c4f2455c9dfa10e4

**Comment by apoelstra** [comment]
I ACKed because my local tests pass, but the real CI fails (an unsurprising outcome for a PR that's trying to change CI and not much else).

**Comment by tcharding** [comment]
Hey @storopoli I've sent you on a wild goose chase, I forgot we do not run the linter in CI in this repo. I think you should throw the `dead_code` attribute on those two functions but @apoelstra something is not right, needs further investigation as to why those to functions aren't used.

You could put this on the attribute: `TODO(Tobin): We shouldn't need this attribute`

**Comment by apoelstra** [comment]
Actually -- if the CI failures are because of missing parts from #638, then I can merge this. But I probably won't have time to check this until tomorrow.

**Comment by tcharding** [comment]
kludge and merge!

**Comment by tcharding** [comment]
What! I don't have overlord powers to change peoples PR descriptions - outrageous. @storopoli can you please remove the mention in the PR description, it makes github spam my notifications. Thanks man.

**Comment by storopoli** [comment]
> What! I don't have overlord powers to change peoples PR descriptions - outrageous. @storopoli can you please remove the mention in the PR description, it makes github spam my notifications. Thanks man.

Done, sorry about the spam.

**Comment by apoelstra** [comment]
> What! I don't have overlord powers to change peoples PR descriptions - outrageous. @storopoli can you please remove the mention in the PR description, it makes github spam my notifications. Thanks man.

Just so you know, the merge script warns about this (and if you can't edit descriptions, you can't merge either, I guess you're not a maintainer here) and I always edit out `@` characters before merging peoples' stuff.

**Comment by apoelstra** [comment]
Lol, CI won't run the MSRV job unless the nightly job also passes. That's a dumb ordering because nightly breaks on its own and MSRV only breaks if the code is wrong.

Anyway I'll just run contrib.sh locally and see how things work.

**Comment by apoelstra** [comment]
Ok, looks good. On the off chance that the pins are actually needed we can reinstate them when we're done other the other CI fixes.

**Pull request merged**
